swaywm/wlroots
# surface: redesign state

- [x] Rename `invalid` to `committed` - better ideas welcome!
- [x] Move `subsurface_position` in `wlr_subsurface`
- [x] Fix `sx`/`sy` handling
- [x] Don't monkey-patch `surface_damage` and `buffer_damage` - add a new buffer damage region instead
- [x] Make `current` and `pending` embedded structs
- [x] Add a `previous` state
- [x] Update cursor hotspot with buffer position

Test plan:
- Make sure damage tracking works on surfaces and subsurfaces (try gnome-calculator)
- Try with scaled and transformed outputs
- Try [wleird](https://github.com/emersion/wleird)'s cursor example

See #1065
